File: eul+/euler0025.cpp
#include <iostream>
#include <cmath>

int find_fib(int);
int num_len(int);

int main() {
    int t, k;
    std::cin >> t;

    for (int i = 0; i != t; ++i) {
        std::cin >> k;
        if (k <= 10) {
            std::cout << find_fib(k) << std::endl;
        } else {
            double n = round(((k - 1) * log(10) + (log(5) / 2)) / log(1.6180));
            std::cout << n << std::endl;
        }
    }
}

int find_fib(int k) {
    int cur = 1, next1 = 1, next2 = 2, n = 1;
    while (num_len(cur) < k) {
        cur = next1;
        next1 = next2;
        next2 = cur + next1;
        ++n;
    }
    return n;
}

int num_len(int n) {
    int len = 0;
    while (n > 0) {
        n /= 10;
        ++len;
    }
    return len;
}

File: eul+/euler0025.py
import math

def num_len(n):
    l = 0
    while (n > 0):
        n /= 10
        l += 1
    return l

def find_fib(k):
    cur = 1
    n1 = 1
    n2 = 2
    n = 1
    while (num_len(cur) < k):
        cur = n1
        n1 = n2
        n2 = cur + n1
        n += 1
    return n

t = input()

for i in range(t):
    k = input()
    if (k <= 10000):
        print find_fib(k)
    
    n = int(math.ceil(((k - 1) * math.log(10) + (math.log(5) / 2)) / math.log(1.6180)))
    print n
